### PR TITLE
Tests: Fix test warning/error caused by updating props in the zustand store

### PR DESCRIPTION
**Pull request by OlegMoshkovich**
This PR address #337.
The warnings were caused by updating the store before render, which is incorrect order, and not awaiting for the update.
- Button Test is still throwing the error,  since we are testing for this exact error - the behavior is correct.
- Side drawer - opened via URL'  still causing a warning - it is commented out for now. 
- Permalink Mock needs more attention

### CODE REVIEW

File: src/Components/IssuesControl.test.jsx [hunk 1]
```diff
@@ -1,57 +1,57 @@
...
 })
 
 
-test('Issues ', () => {
+test('Setting comments in zustand ', async () => {
   const {result} = renderHook(() => useStore((state) => state))
   const testIssueId = 10
-  act(() => {
+  const {getByText} = render(<ShareMock><Issues/></ShareMock>)
+  await act(() => {
```
Comment (OlegMoshkovich) [inline src/Components/IssuesControl.test.jsx:43]: @pablo-mayrgundter this worked to resolve the warning

**Review by OlegMoshkovich — COMMENTED** [review]
@pablo-mayrgundter left a comment

File: src/Components/IssuesControl.test.jsx (hunk 1 above)
Comment (pablo-mayrgundter) [reply src/Components/IssuesControl.test.jsx:43]: Heya, just getting to this.  Thanks but also needs a merge now.  Ogali has a PR for just the await fix that will probably be in soon too.

Comment (OlegMoshkovich) [reply src/Components/IssuesControl.test.jsx:43]: I think Ogali merged his fix.
there are conflicts now. 
shall I rebase?

Comment (pablo-mayrgundter) [reply src/Components/IssuesControl.test.jsx:43]: Yes sure

**Review by pablo-mayrgundter — COMMENTED** [review]
Ok, basically looks good.  Maybe we can chat on Discord or talk?  I'd like to understand the fix better.

High-level question too, looks like you reverted my switch to describe/it?

**Comment by pablo-mayrgundter** [comment]
Per discussion, please use describe/it.  describe/it provides 3 scopes: global, suite and test.  global is top-level statements.. suite allows setup and tear down for this set of tests.  I've been moving us towards that.

**Comment by OlegMoshkovich** [comment]
@pablo-mayrgundter PTAL

**Review by pablo-mayrgundter — APPROVED** [review]

**Pull request merged**